src/App.tsx

```diff
@@ -1,61 +1,121 @@
+import React, { useState } from 'react';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [currentTodos, setCurrentTodos] = useState(todos);
+  const [isUserSelected, setIsUserSelected] = useState(true);
+  const [isTitleEntered, setIsTitleEntered] = useState(true);
+
+  const newId = Math.max(...currentTodos.map(todo => todo.id)) + 1;
+
+  function addTodo(id: number) {
+    if (getUser(id) && title.trim()) {
+      const newTodo = {
+        id: newId,
+        userId,
+        title,
+        completed: false,
+        user: getUser(userId),
+      };
+
+      setUserId(0);
+      setTitle('');
+      setCurrentTodos([...currentTodos, newTodo]);
+    }
+
+    if (!userId) {
+      setUserId(0);
+      setIsUserSelected(false);
+    }
+
+    if (!title.trim()) {
+      setIsTitleEntered(false);
+    }
+  }
+
+  function handleFormSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    addTodo(userId);
+  }
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleFormSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleInput">Title: </label>
+          <input
+            type="text"
+            data-cy="titleInput"
+            id="titleInput"
+            value={title}
+            onChange={(event) => {
+              setTitle(event.target.value);
+              setIsTitleEntered(true);
+            }}
+            placeholder="Enter a title"
+          />
+
+          {!isTitleEntered && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <label htmlFor="userSelect">User: </label>
+          <select
+            data-cy="userSelect"
+            id="userSelect"
+            value={userId}
+            onChange={(event) => {
+              setUserId(Number(event.target.value));
+              setIsUserSelected(true);
+            }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)